src/corelibs/U2Core/src/models/DocumentModel.cpp

```diff
@@ -313,11 +313,13 @@ void Document::_addObject(GObject* obj) {
 bool Document::removeObject(GObject* obj, DocumentObjectRemovalMode removalMode) {
     SAFE_POINT(df->isObjectOpSupported(this, DocumentFormat::DocObjectOp_Remove, obj->getGObjectType()), "Unsupported format operation", false);
 
+    emit si_beforeObjectRemoved(obj);
+
     switch (removalMode) {
         case DocumentObjectRemovalMode_Deallocate:
             return _removeObject(obj, true);
         case DocumentObjectRemovalMode_OnlyNotify:
-            emit si_objectRemoved(obj);
+            emit si_afterObjectRemoved(obj);
```

Comment:
I'm OK with this change, but I only asked to name the new method correctly.
Renaming the old method we need to be very careful to avoid loosing existing SIGNAL/SLOTs.